aqueduct/gateway/views/mcp.py

```diff
@@ -0,0 +1,801 @@
+            log.warning(
+                f"Session {session_id} not ready in POST request (state: {session.state.value})"
+            )
+            return JsonResponse(
+                {"error": f"Session not ready (state: {session.state.value})"}, status=503
+            )
+
+        # Register operation for requests that expect responses
+        operation_registered = False
+        if isinstance(json_rpc_message.root, JSONRPCRequest):
+            session.register_operation_start()
+            operation_registered = True
+
+        try:
+            log.debug(f"Sending message to session {session_id}")
+            session_message = SessionMessage(json_rpc_message)
+            await session_manager.send_message(session_id, session_message)
+
+            # Only wait for response if this is a request (not a notification)
+            # Per MCP spec: "The server MUST NOT send a response to notifications"
+            if isinstance(json_rpc_message.root, JSONRPCRequest):
+                log.debug(f"Waiting for response from session {session_id}")
+                received_message = await session_manager.receive_message(session_id)
+                log.debug(f"Received response from session {session_id}")
+
+                if isinstance(received_message, Exception):
+                    log.error(f"Session {session_id} returned exception: {str(received_message)}")
+                    # Convert exception to JSON-RPC error instead of HTTP error
+                    error_data = {"code": -32001, "message": str(received_message)}
+                    error_response = {
+                        "jsonrpc": "2.0",
+                        "id": json_rpc_message.root.id,
+                        "error": error_data,
+                    }
+                    response = JsonResponse(error_response)
+                    if session_id:
+                        response.headers["mcp-session-id"] = session_id
+                    return response
+
+                response_data = received_message.message.model_dump(exclude_none=True)
+                log.debug(f"MCP response session {session_id}: {len(str(response_data))} chars")
+                response = JsonResponse(response_data)
+                if session_id:
+                    response.headers["mcp-session-id"] = session_id
+                return response
+            else:
+                log.debug(f"Notification processed for session {session_id}")
+                response = JsonResponse({"status": "accepted"}, status=202)
+                if session_id:
+                    response.headers["mcp-session-id"] = session_id
+                return response
+        finally:
+            if operation_registered:
+                session.register_operation_done()
+    except (KeyError, RuntimeError) as e:
+        log.error(f"MCP server '{name}' not found: {str(e)}")
+        return JsonResponse({"error": f"MCP server '{name}' not found: {str(e)}"}, status=404)
```

Comment:
The message suggests that you're expecting some very specific error that is not likely to happen in, say, the part of code responsible for handling the situation when `json_rpc_message.root` is a `JSONRPCRequest`. Can you please try to limit the `try ... except` blocks to only contain the exact lines when you expect the given exception to happen? Especially with multiple nested try-excepts, this code is very hard do read. Also, if a `KeyError` happens anywhere else, it's going to be captured here and logged with a message that may be more misleading than helpful (especially given the fact that `str(e)` will just be the key that was not found).